File: frontend/src/types/financialStatement.ts
export type MoneyValue = string | number;

export interface FinancialStatementGenerateRequest {
  period: string;
  account_set_id?: string;
  operator?: string;
}

export interface StatementLineItem {
  code: string;
  name: string;
  amount: MoneyValue;
  formula: string;
}

export interface BalanceSheet {
  title: string;
  period: string;
  items: StatementLineItem[];
  total_assets: MoneyValue;
  total_liabilities: MoneyValue;
  total_equity: MoneyValue;
  total_liabilities_and_equity: MoneyValue;
  balanced: boolean;
}

export interface IncomeStatement {
  title: string;
  period: string;
  items: StatementLineItem[];
  total_revenue: MoneyValue;
  total_cost: MoneyValue;
  total_expense: MoneyValue;
  total_profit: MoneyValue;
  net_profit: MoneyValue;
}

export interface CashFlowStatement {
  title: string;
  period: string;
  items: StatementLineItem[];
  operating_cash_flow_net: MoneyValue;
  investing_cash_flow_net: MoneyValue;
  financing_cash_flow_net: MoneyValue;
  net_cash_flow: MoneyValue;
}

export interface EquityStatement {
  title: string;
  period: string;
  items: StatementLineItem[];
  opening_equity: MoneyValue;
  current_period_profit: MoneyValue;
  closing_equity: MoneyValue;
}

export interface ManagementStatementSummary {
  title: string;
  key_metrics: Record<string, string>;
  highlights: string[];
  risks: string[];
}

export interface FinancialStatementGenerationSummary {
  account_set_id: string;
  period: string;
  source: string;
  reviewed_voucher_count: number;
  asset_liability_balanced: boolean;
  generated_statement_count: number;
}

export interface FinancialStatementBundle {
  account_set_id: string;
  period: string;
  company_name: string;
  source: string;
  summary: FinancialStatementGenerationSummary;
  balance_sheet: BalanceSheet;
  income_statement: IncomeStatement;
  cash_flow_statement: CashFlowStatement;
  equity_statement: EquityStatement;
  management_summary: ManagementStatementSummary;
}

File: backend/app/models/financial_statement.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.finance import MONTH_PATTERN


class FinancialStatementGenerateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    period: str = Field(pattern=MONTH_PATTERN)
    account_set_id: str = Field(default="default", min_length=1, max_length=64)
    operator: str = Field(default="财务主管", min_length=1, max_length=32)


class StatementLineItem(BaseModel):
    code: str
    name: str
    amount: Decimal
    formula: str


class BalanceSheet(BaseModel):
    title: str
    period: str
    items: list[StatementLineItem]
    total_assets: Decimal
    total_liabilities: Decimal
    total_equity: Decimal
    total_liabilities_and_equity: Decimal
    balanced: bool


class IncomeStatement(BaseModel):
    title: str
    period: str
    items: list[StatementLineItem]
    total_revenue: Decimal
    total_cost: Decimal
    total_expense: Decimal
    total_profit: Decimal
    net_profit: Decimal


class CashFlowStatement(BaseModel):
    title: str
    period: str
    items: list[StatementLineItem]
    operating_cash_flow_net: Decimal
    investing_cash_flow_net: Decimal
    financing_cash_flow_net: Decimal
    net_cash_flow: Decimal


class EquityStatement(BaseModel):
    title: str
    period: str
    items: list[StatementLineItem]
    opening_equity: Decimal
    current_period_profit: Decimal
    closing_equity: Decimal


class ManagementStatementSummary(BaseModel):
    title: str
    key_metrics: dict[str, str]
    highlights: list[str]
    risks: list[str]


class FinancialStatementGenerationSummary(BaseModel):
    account_set_id: str
    period: str
    source: str
    reviewed_voucher_count: int
    asset_liability_balanced: bool
    generated_statement_count: int
    base_currency: str = "CNY"
    foreign_currency_line_count: int = 0


class FinancialStatementBundle(BaseModel):
    account_set_id: str
    period: str
    company_name: str
    source: str
    summary: FinancialStatementGenerationSummary
    balance_sheet: BalanceSheet
    income_statement: IncomeStatement
    cash_flow_statement: CashFlowStatement
    equity_statement: EquityStatement
    management_summary: ManagementStatementSummary
